src/Abstractions/TaskOptions.cs

```diff
@@ -52,6 +52,13 @@ public TaskOptions(TaskRetryOptions? retry = null)
     /// <param name="instanceId">The instance ID to use.</param>
     /// <returns>A new <see cref="SubOrchestrationOptions" />.</returns>
     public SubOrchestrationOptions WithInstanceId(string instanceId) => new(this, instanceId);
+
+    /// <summary>
+    /// Returns a new <see cref="CallActivityOptions" /> with the provided tags.
+    /// </summary>
+    /// <param name="tags">The tags to associate with the activity.</param>
+    /// <returns>A new <see cref="CallActivityOptions" />.</returns>
+    public CallActivityOptions WithTags(IDictionary<string, string> tags) => new(this, tags);

```

Comment:
Returning `CallActivityOptions` means we can't use this for sub-orchestrations in the future, correct? However, I think supporting sub-orchestrations will be important since we support it already for top-level orchestrations. How can we modify the design to support both?

Reply:
moved tags to taskoptions as common optional field since it is common to all subclasses supported currently. suborchoptions can later add new ctor to create with tags. also removed withtags fluent api (not extensible if in future we add more and more fields to subtypes, need to create lots of withXXYYZZ functions) and let customers create callactivityoptions directly or using c9 record with api.